packages/app/src/app/pages/common/Modals/NewTeamModal/TeamMembers.tsx

```diff
@@ -1,25 +1,150 @@
-import React from 'react';
-import { Button, Stack, Textarea, Text } from '@codesandbox/components';
-import { useAppState } from 'app/overmind';
+import React, { FormEvent, useState } from 'react';
+import { Button, Stack, Text } from '@codesandbox/components';
+import { useAppState, useEffects } from 'app/overmind';
+import { StyledButton } from 'app/components/dashboard/Button';
+import { Textarea } from 'app/components/dashboard/Textarea';
+import { TeamMemberAuthorization } from 'app/graphql/types';
+
+function validateEmail(email: string) {
+  // Test for "anything@anything.anything" and check for
+  // multiple @ signs.
+  return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email);
+}
+
+function formatInvalidEmails(invalidEmails: string[]) {
+  const isSingleEmail = invalidEmails.length === 1;
+
+  if (isSingleEmail) {
+    return `${invalidEmails[0]}.`;
+  }
+
+  return invalidEmails.map((invalidEmail, index, arr) => {
+    const isFirstEmail = index === 0;
+    const isLastEmail = arr.length - 1 === index;
+
+    return `${
+      isLastEmail
+        ? ` and ${invalidEmail}.`
+        : `${isFirstEmail ? '' : ', '}${invalidEmail}`
+    }`;
+  });
+}
 
 export const TeamMembers: React.FC<{ onComplete: () => void }> = ({
   onComplete,
 }) => {
   const { activeTeamInfo } = useAppState();
+  const { gql } = useEffects();
+  const [addressesString, setAddressesString] = useState<string>();
+  const [invalidEmails, setInvalidEmails] = useState<string[]>();
+  const [inviteError, setInviteError] = useState<string>();
+
+  const handleSubmit = async (e: FormEvent) => {
+    e.preventDefault();
+    setInviteError(null);
+
+    // Remove spaces (and other whitespace) and split email addresses
+    const emails = addressesString.replace(/\s/g, '').split(',');
+
+    // Validate emails
+    const invalid = emails.filter(email => {
+      const isValidEmail = validateEmail(email);
+
+      // Return when the email is not valid
+      return !isValidEmail;
+    });
+
+    if (invalid.length === 0) {
+      setInvalidEmails(null);
+
+      // Invite via email
+      try {
+        await gql.mutations.inviteToTeamViaEmail({
+          teamId: activeTeamInfo.id,
+          email: addressesString,
+          authorization: TeamMemberAuthorization.Write,
+        });
+
+        onComplete();
+      } catch (error) {
+        // ❗️ TODO: Validate if this works!
+        // Copied logic from inviteToTeam function in dashboard/actions.ts
+        const message =
+          error?.response?.errors?.[0]?.message ||
+          'There was a problem sending the invitations.';
+
+        setInviteError(message);
+      }
```

Comment:
Tried to validate this, but for now all "errors" return `200`. Client side validation makes sure it's properly caught, unless someone willingly turns it off. We don't supply an error for that right now (we would have to check the result of the mutation for a certain string that is susceptible to change).

I would like to keep this logic though, in case we receive a 500 or any other error. What do you think?